plugins/woocommerce/includes/class-wc-brands.php

```diff
@@ -237,10 +237,70 @@ public function add_body_class( $classes ) {
 	 * Enqueues styles.
 	 */
 	public function styles() {
+		if ( ! $this->should_load_brands_styles() ) {
+			return;
+		}
+
 		$version = Constants::get_constant( 'WC_VERSION' );
 		wp_enqueue_style( 'brands-styles', WC()->plugin_url() . '/assets/css/brands.css', array(), $version );
 	}
 
+	/**
+	 * Determines if brands styles should be loaded on the current page.
+	 *
+	 * @since 10.4.0
+	 * @return bool
+	 */
+	private function should_load_brands_styles() {
+		global $post;
+
+		// Should load on brand taxonomy archive pages.
+		if ( is_tax( 'product_brand' ) ) {
+			return true;
+		}
+
+		// Should load on single product pages that have brands assigned.
+		if ( is_singular( 'product' ) && $post ) {
+			$terms = get_the_terms( $post->ID, 'product_brand' );
+			if ( ! empty( $terms ) && ! is_wp_error( $terms ) ) {
+				return true;
+			}
+		}
+
+		// Check if any brand shortcodes are present in the content.
+		if ( $post && isset( $post->post_content ) ) {
+			$brand_shortcodes = array(
+				'brand_products',
+				'product_brand',
+				'product_brand_list',
+				'product_brand_thumbnails',
+				'product_brand_thumbnails_description',
+			);
+
+			foreach ( $brand_shortcodes as $shortcode ) {
+				if ( has_shortcode( $post->post_content, $shortcode ) ) {
+					return true;
+				}
+			}
+		}
+
+		// Check if any brand widgets are active.
+		if ( is_active_widget( false, false, 'woocommerce_brand_description' ) ||
+			is_active_widget( false, false, 'woocommerce_brand_nav' ) ||
+			is_active_widget( false, false, 'woocommerce_brand_thumbnails' ) ) {
```

Comment:
It's called `wc_brands_brand_thumbnails` as per this:

https://github.com/woocommerce/woocommerce/blob/d821a5778e54633a614a25bdeab164e827e70b56/plugins/woocommerce/includes/widgets/class-wc-widget-brand-thumbnails.php#L51

```suggestion
			is_active_widget( false, false, 'wc_brands_brand_thumbnails' ) ) {
```